source/client/options_impl.cc

```diff
@@ -672,11 +672,11 @@ CommandLineOptionsPtr OptionsImpl::toCommandLineOptionsInternal() const {
     }
   }
 
-  // Only set the tls context if it looks non-empty, to avoid a warning being logged about field
-  // deprecation. Ideally this would follow the way transport_socket uses absl::optional below.
+  // Only set the tls context if needed, to avoid a warning being logged about field deprecation.
+  // Ideally this would follow the way transport_socket uses absl::optional below.
   // But as this field is about to get eliminated this minimal effort shortcut may be more suitable.
-  if (!Envoy::MessageUtil()(tls_context_,
-                            envoy::extensions::transport_sockets::tls::v3::UpstreamTlsContext())) {
+  if (!Envoy::Protobuf::util::MessageDifferencer::Equivalent(
```

Comment:
Note - equivalent is not the same as equal. This would say that two messages are the same if one has a value set to zero and the other one has it unset where the default is zero.

Considering we are using this to determine if the message is unset, is that behavior desired?

Reply:
That is a good catch, thanks. Thinking this through, you're right, this isn't what we need here. I switched this to use the  `ByteSizeLong()` method you pointed out. I'm not super concerned about it not being the fastest way to pull this off, as it is going to be eliminated at some point. Plus this isn't in the hot path. Addressed in f467e86